parquet-column/src/main/java/org/apache/parquet/schema/PrimitiveStringifier.java

```diff
@@ -309,48 +309,89 @@ Instant getInstant(long value) {
     }
   };
 
-  static final PrimitiveStringifier TIME_STRINGIFIER = new PrimitiveStringifier("TIME_STRINGIFIER") {
+  static final PrimitiveStringifier TIMESTAMP_MILLIS_UTC_STRINGIFIER = new DateStringifier(
+    "TIMESTAMP_MILLIS_UTC_STRINGIFIER", "yyyy-MM-dd'T'HH:mm:ss.SSSZ") {
     @Override
-    public String stringify(int millis) {
-      return toTimeString(millis, MILLISECONDS);
+    Instant getInstant(long value) {
+      return Instant.ofEpochMilli(value);
     }
+  };
 
+  static final PrimitiveStringifier TIMESTAMP_MICROS_UTC_STRINGIFIER = new DateStringifier(
+    "TIMESTAMP_MICROS_UTC_STRINGIFIER", "yyyy-MM-dd'T'HH:mm:ss.SSSSSSZ") {
     @Override
-    public String stringify(long micros) {
-      return toTimeString(micros, MICROSECONDS);
+    Instant getInstant(long value) {
+      return Instant.ofEpochSecond(MICROSECONDS.toSeconds(value), MICROSECONDS.toNanos(value % SECONDS.toMicros(1)));
```

Comment:
I think this will lead to incorrect results with negative numbers.

Reply:
@zivanfi I extended the test cases to run on both UTC and non-UTC stringifiers, and they pass. Could you please give some hints which values could be problematic?

Reply:
I thought that negative numbers (representing timestamps before 1970-01-01 00:00:00) would behave incorrectly because of splitting to whole seconds plus nanoseconds part using the % operator. For example, 1.2 would split into 1 + 0.2, which is good, but -1.2 would split into -2 + 0.8 instead of -1 + -0.2. However, based on your test it works correctly, so it seems that I was mistaken (although I still don't see why it works).